### PR TITLE
[SPA] Update to latest oidc-version and perform appropriate cleanups

### PR DESCRIPTION
**Pull request by javiercn**
Updates to the latest version of oidc-client and removes a bunch of template code as a result.

### CODE REVIEW

**Comment by javiercn** [comment]
I plan to merge this as part of **preview9**, but getting it done early in the milestone.

**Review by ryanbrandenburg — APPROVED** [review]
:shipit: if the tests pass.

**Comment by leecow** [comment]
tactics approved - wait to merge for @wtgodbe go ahead.

**Comment by brockallen** [comment]
@javiercn let me know when you're happy and i can do the formal release of oidc-client.

**Comment by wtgodbe** [comment]
You don't need to wait on me for this PR, that's only for CoreFx/Core-Setup/CoreClr

File: eng/Baseline.Designer.props [hunk 1]
```diff
@@ -31,19 +31,6 @@
...
-  </PropertyGroup>
-  <ItemGroup Condition=" '$(PackageId)' == 'Microsoft.AspNetCore.ApiAuthorization.IdentityServer' AND '$(TargetFramework)' == 'netstandard2.0' ">
-    <BaselinePackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="[2.2.0, )" />
-    <BaselinePackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="[2.2.0, )" />
-    <BaselinePackageReference Include="IdentityServer4" Version="[2.3.0-preview1-update2, )" />
-    <BaselinePackageReference Include="IdentityServer4.AspNetIdentity" Version="[2.3.0-preview1-update2, )" />
-    <BaselinePackageReference Include="IdentityServer4.EntityFramework" Version="[2.3.0-preview1-update1, )" />
-    <BaselinePackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="[2.2.0, )" />
-    <BaselinePackageReference Include="Microsoft.AspNetCore.Mvc" Version="[2.2.0, )" />
-  </ItemGroup>
```
Comment (dougbu) [inline eng/Baseline.Designer.props]: Revert this change.

File: eng/Baseline.xml [hunk 2]
```diff
@@ -10,7 +10,6 @@ build of ASP.NET Core 2.2.x. Update this list when preparing for a new patch.
   <Package Id="dotnet-user-secrets" Version="2.2.0" />
   <Package Id="dotnet-watch" Version="2.2.0" />
   <Package Id="Microsoft.AspNetCore.Antiforgery" Version="2.2.0" />
-  <Package Id="Microsoft.AspNetCore.ApiAuthorization.IdentityServer" Version="2.2.0-preview-35687" />
```
Comment (dougbu) [inline eng/Baseline.xml]: Revert this change

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/Angular-CSharp.csproj.in [hunk 3]
```diff
@@ -20,6 +20,7 @@
     <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="${MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' " />
     <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="${MicrosoftAspNetCoreIdentityEntityFrameworkCorePackageVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' " />
     <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="${MicrosoftAspNetCoreIdentityUIPackageVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' " />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="${MicrosoftEntityFrameworkCoreRelationalPackageVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' AND '$(UseLocalDB)' == 'True'" />
```
Comment (dougbu) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/Angular-CSharp.csproj.in]: This shouldn't be needed because Microsoft.EntityFrameworkCore.SqlServer brings it in. What problem does it address (here and elsewhere in this PR)?

File: src/ProjectTemplates/test/SpaTemplateTest/SpaTemplateTestBase.cs [hunk 4]
```diff
@@ -83,7 +82,7 @@ public class SpaTemplateTestBase : BrowserTestBase
             Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("build", Project, buildResult));
 
             // localdb is not installed on the CI machines, so skip it.
-            var shouldVisitFetchData = !useLocalDb;
+            var shouldVisitFetchData = !(useLocalDb && Project.IsCIEnvironment);
```
Comment (dougbu) [inline src/ProjectTemplates/test/SpaTemplateTest/SpaTemplateTestBase.cs]: shouldn't `&&` be `||`?

**Review by dougbu — COMMENTED** [review]
Not seeing anything obvious that would break validation builds. Looking more&hellip;

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/Angular-CSharp.csproj.in (hunk 3 above)
Comment (javiercn) [reply src/ProjectTemplates/Web.Spa.ProjectTemplates/Angular-CSharp.csproj.in]: We need to reference this explicitly because the RTM version of Identity Server will reference preview9 bits of identity server and we want to ensure the version is lift up.

File: src/ProjectTemplates/test/SpaTemplateTest/SpaTemplateTestBase.cs (hunk 4 above)
Comment (javiercn) [reply src/ProjectTemplates/test/SpaTemplateTest/SpaTemplateTestBase.cs]: No, we want this to run locally, but not on the CI

File: eng/Baseline.xml (hunk 2 above)
Comment (javiercn) [reply eng/Baseline.xml]: Why do we keep this if we didn't ship this.

File: eng/Baseline.Designer.props (hunk 1 above)
Comment (javiercn) [reply eng/Baseline.Designer.props]: Same comment as below.

I can revert these two, but it doesn't matter.

Comment (dougbu) [reply eng/Baseline.Designer.props]: Thanks for doing this. It matters because the 'release/3.0' branch does not control the baseline files and changing them here will cause merge conflicts.

File: src/ProjectTemplates/test/SpaTemplateTest/SpaTemplateTestBase.cs (hunk 4 above)
Comment (dougbu) [reply src/ProjectTemplates/test/SpaTemplateTest/SpaTemplateTestBase.cs]: I may be missing something but the expression currently means data will be fetched in CI builds when `!useLocalDb`.

File: src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj [hunk 5]
```diff
@@ -5,19 +5,29 @@
     <Description>ASP.NET Core API Authorization package powered by Identity Server.</Description>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <PackageTags>aspnetcore;apiauth;identity</PackageTags>
+    <HasReferenceAssembly>false</HasReferenceAssembly>
     <IsShippingPackage>true</IsShippingPackage>
+    <HasReferenceAssembly>false</HasReferenceAssembly>
```
Comment (dougbu) [inline src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj]: Set this once (dupe of line 8). As long as our policies allow us to live w/o a ref/ assembly, wonderful 😃

File: src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj [hunk 6]
```diff
@@ -5,19 +5,29 @@
...
     <Reference Include="IdentityServer4.AspNetIdentity" />
     <Reference Include="IdentityServer4.EntityFramework" />
     <Reference Include="IdentityServer4.EntityFramework.Storage" />
     <Reference Include="IdentityServer4" />
     <Reference Include="IdentityServer4.Storage" />
-    <Reference Include="Microsoft.AspNetCore.Authentication.JwtBearer" />
     <Reference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" />
     <Reference Include="Microsoft.AspNetCore.Identity.UI" />
+
+    <!-- These references don't matter as they'll be replaced by the shared framework reference -->
```
Comment (dougbu) [inline src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj]: Not sure this sentence helps much. In any case, it's inconsistent w/ the other projects and you've described the problem already just a few lines earlier.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/package-lock.json [hunk 7]
```diff
@@ -7153,10 +7157,13 @@
...
-      "version": "1.9.0-beta.3",
-      "resolved": "https://registry.npmjs.org/oidc-client/-/oidc-client-1.9.0-beta.3.tgz",
-      "integrity": "sha512-8JqmesN7n7sQrdeEYiuWyWjNp/6cZGOHU5IaYrULm/qdYo6cAheeZl93bDkob6MdfbQMsQ2HGG/+EOHFdFQcSQ==",
+      "version": "1.9.0-beta.4",
+      "resolved": "https://registry.npmjs.org/oidc-client/-/oidc-client-1.9.0-beta.4.tgz",
+      "integrity": "sha512-BE6o220w9kDTicmsX324Z5t3Gg20G9NgQjCOFP6SUY7glJYBohyw9DVPt/khPmmg+wabufJq1tH2zyEadMf7Pw==",
       "requires": {
+        "base64-js": "^1.3.0",
+        "core-js": "^2.6.4",
+        "crypto-js": "^3.1.9-1",
```
Comment (dougbu) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/package-lock.json]: Just to confirm: @Pilchie has approved the added transitive package dependencies (and increased visibility of the base64-js dependency) too?

**Review by dougbu — APPROVED** [review]
Nice
